main.tf

```diff
@@ -104,3 +104,14 @@ data "aws_region" "this" {
 data "aws_caller_identity" "this" {
   count = local.enabled ? 1 : 0
 }
+
+resource "aws_lambda_event_source_mapping" "trigger" {
+  count = local.enabled && var.source_mapping_enabled ? 1 : 0
```

Comment:
please enable `var.source_mapping_enabled` in the complete example so that terratest will validate this resource

Reply:
Hi @milldr can you check if I've got it correctly? I just added the `var.source_mapping_enabled` (whhich defaults to `false`) `examples/variables.tf` and added the new var to main.tf. Is that what was needed?

Reply:
Yup it's as easy as that! For more complex resources we'd like to create an output and validate that output in terratest itself. But for this simple case, a smoke test should be enough

However can you set `source_mapping_enabled` to `true` for the example fixtures only? So that they are spun up with the example in terratest
https://github.com/cloudposse/terraform-aws-lambda-function/blob/main/examples/complete/fixtures.us-east-2.tfvars

Reply:
the idea with `complete` example is that everything is enabled, so we can have some level of basic validation

Reply:
If I set that to true, shouldn't I also set the `source_mapping_arn` variable?

Reply:
the [value in the variables.tf ](https://github.com/cloudposse/terraform-aws-lambda-function/blob/main/examples/complete/variables.tf) is the default value for the example. So it really doesn't matter if you set the value in fixtures too - since it's overwritten

Reply:
you could set the default value to `true` and not add the fixture, but generally as convention we set the value we want to test that is different than the module's default with fixtures

Reply:
Added `source_mapping_enabled = true` in fixtures